test/e2e/tests/test_adopted_resource.py

```diff
@@ -0,0 +1,201 @@
+    _, _ = sagemaker_make_endpoint(endpoint_name, endpoint_config_name)
+
+    wait_sagemaker_endpoint_status(endpoint_name, ENDPOINT_STATUS_INSERVICE)
+    assert resource_id is not None
+
+    yield resource_id
+    sagemaker_client().delete_endpoint(EndpointName=endpoint_name)
+    sagemaker_client().delete_endpoint_config(EndpointConfigName=endpoint_config_name)
+    sagemaker_client().delete_model(ModelName=model_name)
+
+
+@pytest.fixture(scope="module")
+def register_scalable_target(sagemaker_endpoint):
+    resource_id = sagemaker_endpoint
+    _ = sagemaker_endpoint_register_scalable_target(resource_id)
+
+    yield resource_id
+    sagemaker_endpoint_deregister_scalable_target(resource_id)
+
+
+@pytest.fixture(scope="module")
+def put_scaling_policy(register_scalable_target):
+    resource_id = register_scalable_target
+    policy_name = random_suffix_name("policy_name", 32)
+    _ = sagemaker_endpoint_put_scaling_policy(resource_id, policy_name)
+
+    yield resource_id
+    sagemaker_endpoint_delete_scaling_policy(resource_id, policy_name)
+
+
+@pytest.fixture(scope="module")
+def adopt_scalable_target(sagemaker_endpoint):
+    resource_id = sagemaker_endpoint
+    target_resource_name = random_suffix_name("sagemaker-scalable-target", 32)
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["ADOPTED_TARGET_NAME"] = target_resource_name
+    replacements["RESOURCE_ID"] = resource_id
+
+    adopted_target_reference, _, adopted_target_resource = create_adopted_resource(
+        resource_name=target_resource_name,
+        spec_file="sagemaker_endpoint_adopted_target",
+        replacements=replacements,
+    )
+
+    assert adopted_target_resource is not None
+
+    yield (resource_id, adopted_target_reference)
+
+    if k8s.get_resource_exists(adopted_target_reference):
+        _, deleted = k8s.delete_custom_resource(adopted_target_reference)
+        assert deleted
+
+
+@pytest.fixture(scope="module")
+def adopt_scaling_policy(adopt_scalable_target):
+    resource_id, adopted_target_reference = adopt_scalable_target
+    policy_resource_name = random_suffix_name("sagemaker-scaling-policy", 32)
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["ADOPTED_POLICY_NAME"] = policy_resource_name
+    replacements["RESOURCE_ID"] = resource_id
+
+    adopted_policy_reference, _, adopted_policy_resource = create_adopted_resource(
+        resource_name=policy_resource_name,
+        spec_file="sagemaker_endpoint_adopted_policy",
+        replacements=replacements,
+    )
+
+    assert adopted_policy_resource is not None
+
+    yield (resource_id, adopted_target_reference, adopted_policy_reference)
+
+    if k8s.get_resource_exists(adopted_policy_reference):
+        _, deleted = k8s.delete_custom_resource(adopted_policy_reference)
+        assert deleted
+
+
+@service_marker
+@pytest.mark.canary
+class TestAdoptedSageMakerEndpointAutoscaling:
```

Comment:
Please rename to `TestAdoption` and rename the test below to `test_sagemaker_endpoint_autoscaling`. In the future we may add adoption tests for other resources.

Reply:
done.